pkg/remote/client.go

```diff
@@ -0,0 +1,150 @@
+package remote
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/influxdata/influxdb-client-go/v2/api/write"
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+const (
+	NodeIDTag  = "node_id"
+	ChainIDTag = "chain_id"
+)
+
+// EventCollectorConfig is the influxdb client configuration used for
+// collecting events.
+type EventCollectorConfig struct {
+	// URL is the influxdb url.
+	URL string `mapstructure:"influx_url"`
+	// Token is the influxdb token.
+	Token string `mapstructure:"influx_token"`
+	// Org is the influxdb organization.
+	Org string `mapstructure:"influx_org"`
+	// Bucket is the influxdb bucket.
+	Bucket string `mapstructure:"influx_bucket"`
+	// BatchSize is the number of points to write in a single batch.
+	BatchSize int `mapstructure:"influx_batch_size"`
+}
+
+// DefaultEventCollectorConfig returns the default configuration.
+func DefaultEventCollectorConfig() EventCollectorConfig {
+	return EventCollectorConfig{
+		URL:       "",
+		Org:       "celestia",
+		Bucket:    "e2e",
+		BatchSize: 10,
+	}
+}
+
+// Client is an influxdb client that can be used to push events to influxdb. It
+// is used to collect trace data from many different nodes in a network. If
+// there is no URL in the config.toml, then the underlying client is nil and no
+// points will be written. The provided chainID and nodeID are used to tag all
+// points. The underlying client is exposed to allow for custom writes, but the
+// WritePoint method should be used for most cases, as it enforces the schema.
+type Client struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	cfg    *config.InstrumentationConfig
+
+	// chainID is added as a tag all points
+	chainID string
+
+	// nodeID is added as a tag all points
+	nodeID string
+
+	// Client is the influxdb client. This field is nil if no connection is
+	// established.
+	Client influxdb2.Client
+}
+
+// Stop closes the influxdb client.
+func (c *Client) Stop() {
+	c.cancel()
+	if c.Client == nil {
+		return
+	}
+	writeAPI := c.Client.WriteAPI(c.cfg.InfluxOrg, c.cfg.InfluxBucket)
+	writeAPI.Flush()
+	c.Client.Close()
+}
```

Comment:
Should there be also a `Start` method or is this integrated into the constructor?

Reply:
all ongoing processes are handled by the influxdb client. When we create the client we are pinging the server if there is a URL configured. Although I'm not sure if that actually begins any loop, or if the process is started at the first write.

Reply:
Yeah, I'm not sure what exactly the `Close` does (whether it flushes out pending writes or tells the server that it's closing)